pkg/operator/controller/credentials-secret/controller.go

```diff
@@ -143,6 +155,42 @@ func New(mgr manager.Manager, config Config) (controller.Controller, error) {
 		return requests
 	}
 
+	if err := mgr.GetFieldIndexer().IndexField(
+		context.Background(),
+		&operatorv1alpha1.ExternalDNS{},
+		credentialsSecretIndexFieldNameInOperand,
+		client.IndexerFunc(func(o client.Object) []string {
+			ed := o.(*operatorv1alpha1.ExternalDNS)
+			name := "external-dns-credentials-" + ed.Name
```

Comment:
You can use the function from `controller` package to get the name of the secret from the target namespace: [link](https://github.com/openshift/external-dns-operator/blob/main/pkg/operator/controller/names.go#L52) (operand namespace won't matter here).
If you want [the concatenation logic](https://github.com/openshift/external-dns-operator/blob/main/pkg/operator/controller/names.go#L55) can even be put into a new dedicated function.